src/libs/antares/study/version.cpp

```diff
@@ -103,31 +85,52 @@ StudyVersion::StudyVersion(const std::string& s)
         minor = 0;
         logs.error() << "Invalid version format, exiting";
     }
+    return StudyVersion(major, minor);
+}
+
+bool StudyVersion::fromString(const std::string& versionStr)
+{
+    // if the string doesn't contains a dot it's legacy format
+    if (versionStr.find(".") == std::string::npos)
+        *this = parseLegacyVersion(versionStr);
+    else
+        *this = parseCurrentVersion(versionStr);
```

Comment:
```suggestion
    if (size_t pos = versionStr.find("."); pos == std::string::npos)
        *this = parseLegacyVersion(versionStr);
    else
        *this = parseCurrentVersion(pos, versionStr);
```

So you don't need to call find(".") twice